plugin/src/main/java/org/opensearch/ml/action/model_group/TransportUpdateModelGroupAction.java

```diff
@@ -0,0 +1,147 @@
+ */
+
+package org.opensearch.ml.action.model_group;
+
+import static org.opensearch.ml.common.CommonValue.ML_MODEL_GROUP_INDEX;
+import static org.opensearch.ml.utils.MLExceptionUtils.logException;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import lombok.extern.log4j.Log4j2;
+
+import org.apache.commons.lang3.StringUtils;
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.get.GetRequest;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.HandledTransportAction;
+import org.opensearch.action.update.UpdateRequest;
+import org.opensearch.client.Client;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.commons.authuser.User;
+import org.opensearch.core.xcontent.NamedXContentRegistry;
+import org.opensearch.ml.common.MLModelGroup;
+import org.opensearch.ml.common.exception.MLException;
+import org.opensearch.ml.common.transport.model_group.MLUpdateModelGroupAction;
+import org.opensearch.ml.common.transport.model_group.MLUpdateModelGroupInput;
+import org.opensearch.ml.common.transport.model_group.MLUpdateModelGroupRequest;
+import org.opensearch.ml.common.transport.model_group.MLUpdateModelGroupResponse;
+import org.opensearch.ml.utils.RestActionUtils;
+import org.opensearch.ml.utils.SecurityUtils;
+import org.opensearch.tasks.Task;
+import org.opensearch.transport.TransportService;
+
+import com.google.common.collect.ImmutableList;
+
+@Log4j2
+public class TransportUpdateModelGroupAction extends HandledTransportAction<ActionRequest, MLUpdateModelGroupResponse> {
+
+    private final TransportService transportService;
+    private final ActionFilters actionFilters;
+    private Client client;
+    private NamedXContentRegistry xContentRegistry;
+
+    @Inject
+    public TransportUpdateModelGroupAction(
+        TransportService transportService,
+        ActionFilters actionFilters,
+        Client client,
+        NamedXContentRegistry xContentRegistry
+    ) {
+        super(MLUpdateModelGroupAction.NAME, transportService, actionFilters, MLUpdateModelGroupRequest::new);
+        this.actionFilters = actionFilters;
+        this.transportService = transportService;
+        this.client = client;
+        this.xContentRegistry = xContentRegistry;
+    }
+
+    @Override
+    protected void doExecute(Task task, ActionRequest request, ActionListener<MLUpdateModelGroupResponse> listener) {
+        MLUpdateModelGroupRequest updateModelGroupRequest = MLUpdateModelGroupRequest.fromActionRequest(request);
+        MLUpdateModelGroupInput updateModelGroupInput = updateModelGroupRequest.getUpdateModelGroupInput();
+
+        String modelGroupId = updateModelGroupInput.getModelGroupID();
+        GetRequest getModelGroupRequest = new GetRequest(ML_MODEL_GROUP_INDEX).id(modelGroupId);
+
+        if (modelGroupId != null) {
```

Comment:
Use reverse check to eliminate the nested ifs, e.g. `if (modelGroupId == null) throw new IllgalArgumentException(xxx)`